src/glb-director-xdp/main.go

```diff
@@ -331,15 +344,31 @@ func (app *Application) ReloadForwardingTable() {
 		for b := uint32(0); b < uint32(srcTable.num_binds); b++ {
 			bindEntry := srcTable.binds[b]
 
-			bindKey := C.glb_bind{ipv4: C._get_bind_ipv4(&bindEntry), ipv6: C._get_bind_ipv6(&bindEntry), proto: C.ushort(bindEntry.proto), port: C._get_bind_port(&bindEntry)}
-			fmt.Printf("      bind: %v\n", bindKey)
-			tableIndex := uint32(i)
-			if err := tableMap4.Put(unsafe.Pointer(&bindKey), unsafe.Pointer(&tableIndex)); err != nil {
-				log.Fatal(err)
+			bindRange := uint32(C._is_bind_range(&bindEntry))
+			isBindRange := bindRange != 0
+			if isBindRange {
```

Comment:
Happy to remove this bool conversion and just use the uint32 in the comparison

Reply:
I _think_ because in the case where we're using a single port we have `port_start == port_end`, we might be able to always use this loop-based approach and have it work for both ranges and single port binds. Was there anything preventing this?

Reply:
> Was there anything preventing this?

Maybe the difference in the json format and binary format of the binds structure :) but mostly me. And yeah, good catch 👍  Will also eliminate the need for that `_is_bind_range()` cgo function.